source/MaterialXRuntime/Commands/PrimCommands.cpp

```diff
@@ -0,0 +1,287 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRuntime/Commands/PrimCommands.h>
+#include <MaterialXRuntime/Commands/ConnectionCommands.h>
+#include <MaterialXRuntime/RtApi.h>
+
+namespace MaterialX
+{
+
+RtCommandPtr RtCreatePrimCmd::create(RtStagePtr stage, const RtToken& typeName, const RtPath& parentPath, const RtToken& name)
+{
+    return std::make_shared<RtCreatePrimCmd>(stage, typeName, parentPath, name);
+}
+
+void RtCreatePrimCmd::execute(RtCommandResult& result)
+{
+    try
+    {
+        _prim = _stage->createPrim(_parentPath, _name, _typeName);
+        
+        // Update the name if it was changed so we can undo later.
+        _name = _prim.getName();
+
+        result = RtCommandResult(_prim.asA<RtObject>());
+    }
+    catch (const ExceptionRuntimeError& e)
+    {
+        result = RtCommandResult(false, string(e.what()));
+    }
+}
+
+void RtCreatePrimCmd::undo(RtCommandResult& result)
+{
+    try
+    {
+        RtPath path(_parentPath);
+        path.push(_name);
+        _stage->disposePrim(path);
+        result = RtCommandResult(true);
+    }
+    catch (const ExceptionRuntimeError& e)
+    {
+        result = RtCommandResult(false, string(e.what()));
+    }
+}
+
+void RtCreatePrimCmd::redo(RtCommandResult& result)
+{
+    try
+    {
+        _stage->revivePrim(_parentPath, _prim);
```

Comment:
Nitpick: Sorry it sounds like "revive" means we "killed" the prim :).
Maybe restorePrim() to match disposePrim() ?

Reply:
That's a better name, thanks! :)